sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeDirectoryClient.cs

```diff
@@ -3059,11 +3059,170 @@ public override Uri GenerateSasUri(DataLakeSasBuilder builder, out string string
             // Deep copy of builder so we don't modify the user's original DataLakeSasBuilder.
             builder = DataLakeSasBuilder.DeepCopy(builder);
 
+            SetBuilderAndValidate(builder);
+            DataLakeUriBuilder sasUri = new DataLakeUriBuilder(Uri)
+            {
+                Sas = builder.ToSasQueryParameters(ClientConfiguration.SharedKeyCredential, out stringToSign)
+            };
+            return sasUri.ToUri();
+        }
+        #endregion
+
+        #region GenerateUserDelegationSas
```

Comment:
Why did we need to modify `DataLakeDirectoryClient` but not `DataLikeFileClient`?  I am guessing that `DataLakeFileClient` uses the `DataLakePathClient`'s `GenerateUserDelegationSas()` implementation, but `DataLakeDirectoryClient` needs to set `IsDirectory = true`?

Reply:
Good question, and yeah that is pretty much the reasoning. 

DataLakePathClient's GenerateUserDelegationSas() fits perfectly with DataLakeFileClient so we didn't need to modify DataLakeFileClient. But DataLakeDirectoryClient needed some changes (mostly IsDirectory = true) so it was modified.

This pattern is also present for the existing GenerateSas().

Reply:
@amnguye keep me honest here

Reply:
Yup it's cause it needs to set `IsDirectory = true` because Directory SAS is a thing.

This is also following the already existing `GenerateSas(..)` methods on `DataLakeDirectoryClient`